include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1174,11 +1174,6 @@ __parallel_find_or(oneapi::dpl::__internal::__device_backend_tag, _ExecutionPoli
     __wgroup_size = ::std::min(__wgroup_size, oneapi::dpl::__internal::__kernel_work_group_size(__exec, __kernel));
 #endif
 
-#if _ONEDPL_FPGA_EMU
-    // Limit the maximum work-group size to minimize the cost of work-group reduction.
-    // Limiting this also helps to avoid huge work-group sizes on some devices (e.g., FPGU emulation).
-    __wgroup_size = std::min(__wgroup_size, (std::size_t)2048);
```

Comment:
Does replacing this 2048 with 8192 reintroduce any detectable slowdown for CPU / FPGA?

I ask because I believe most codes are written with a max of 1024 mind (with the priority intended target being GPU).  

It would be interesting to see how different kernels in oneDPL perform on CPU / FPGA with 1024, 2048, 4096, 8192.  If in reality we are not getting any "benefit" out of more than 1024 / 2048, I think I would prefer that we use those numbers as the max.  

My suspicion is that sycl group algorithms, and our kernels in general perform worse with these higher workgroup sizes in practice.  

This could be made into an issue if we don't want to address it here.

Reply:
There is a huge range of CPU and FPGA device, but I've collected data on 5th gen Xeon as a reference. The `find_or` pattern and radix sort, scan, reduce-by-segment, and scan-by-segment algorithms use `__max_work_group_size` and/or `__kernel_work_group_size` without lowering the work-group size. Runs with 1024, 2048, 4096, 8192 didn't show a general pattern. Instead, the optimal work-group size is dependent on the specific algorithm. I think our assumption about communication-itensive kernels such as scan benefit from smaller work-group sizes holds though. However, radix sort seems to favor larger sizes. Thus, I've added specific limits based on the empirical data.

Reply:
Very interesting. Thanks for taking a look at this. 
I'm not sure how to feel about these magic values since like you mention, there is a large search space of device.  However, I don't really know a better way.

Reply:
I agree. I think one way to help with these tuning parameters would be to explicitly mark them and provide tooling to reevaluate them based on hardware changes. This would also help improve with some of the tuning work such as https://github.com/oneapi-src/oneDPL/pull/1054.